src/types/web-resource.ts

```diff
@@ -0,0 +1,108 @@
+import * as TypeUtils from '../type-utils';
+import { storageRegistry } from '../storage-adapters/index';
+
+/**
+ * This is how WebResources are stored on the DB
+ */
+export type WebResourceRef = {
+	filename: string;
+	href: string;
+	contentType?: string;
+	contentDisposition?: string;
+	size?: number;
+};
+
+/**
+ * This is how WebResources are defined on creation
+ */
+export type WebResourceInput = {
+	filename: string;
+	data: Buffer;
+	contentType?: string;
+	contentDisposition?: string;
+	size?: number;
+	storage: string;
+};
+
+export const types = {
+	postgres: 'JSONB',
+	mysql: 'JSON',
+	websql: 'TEXT',
+	odata: {
+		name: 'Self.WebResource',
+		complexType: `\
+<ComplexType Name="WebResource">
+	<Property Name="filename" Nullable="false" Type="Edm.String"/>\
+	<Property Name="href" Nullable="false" Type="Edm.String"/>\
+	<Property Name="contentType" Nullable="true" Type="Edm.String"/>\
+	<Property Name="contentDisposition" Nullable="true" Type="Edm.String"/>\
+	<Property Name="size" Nullable="true" Type="Edm.Int64"/>\
+</ComplexType>`,
+	},
+};
+
+export const fetchProcessing = (data: any) => {
+	let refData: WebResourceRef;
+	if (data === null) {
+		return data;
+	}
+	if (typeof data === 'string') {
+		refData = JSON.parse(data);
```

Comment:
do we want to handle errors thrown by JSON.parse here, or where are they handled?

Reply:
Any `SyntaxError` will be propagated. I can catch them and return a properly formatted error as Page suggested above

Reply:
Fixed on [web-resource-2 848c856] Enhance error handling